BUILD

```diff
@@ -0,0 +1,152 @@
+load("@protobuf_git//:protobuf.bzl", "cc_proto_library")
+
+exports_files(["source/precompiled/precompiled.h"])
+
+package(default_visibility = ["//visibility:public"])
+
+genrule(
+    name = "envoy-ratelimit-proto",
+    srcs = [
+        "source/common/ratelimit/ratelimit.proto",
+    ],
+    outs = [
+        "source/common/generated/ratelimit.proto",
+    ],
+    cmd = "cp $(SRCS) $@",
+)
+
+cc_proto_library(
+    name = "envoy-ratelimit-pb",
+    srcs = [
+        "source/common/generated/ratelimit.proto",
+    ],
+    default_runtime = "//external:protobuf",
+    protoc = "//external:protoc",
+    include = "source",
+)
+
+genrule(
+    name = "envoy-test-proto",
+    srcs = [
+        "test/proto/helloworld.proto",
+    ],
+    outs = [
+        "test/generated/helloworld.proto",
+    ],
+    cmd = "cp $(SRCS) $@",
+)
+
+cc_proto_library(
+    name = "envoy-test-pb",
+    srcs = [
+        "test/generated/helloworld.proto",
+    ],
+    default_runtime = "//external:protobuf",
+    protoc = "//external:protoc",
+    include = "test",
+)
+
+genrule(
+    name = "envoy-version",
+    srcs = glob([
+        ".git/**",
+    ]),
+    tools = [
+        "tools/gen_git_sha.sh",
+    ],
+    outs = [
+        "source/common/version_generated.cc",
+    ],
+    cmd = "touch $@ && $(location tools/gen_git_sha.sh) $$(dirname $(location tools/gen_git_sha.sh)) $@",
+    local = 1,
+)
+
+cc_library(
+    name = "envoy-common",
+    srcs = glob([
+        "source/**/*.cc",
+        "source/**/*.h",
+        "include/**/*.h",
+    ], exclude=["source/exe/main.cc"]) + [
+        "source/common/version_generated.cc",
+    ],
```

Comment:
Buildifier: accorduing to `buildifier`, this should be written like this:

    srcs = glob(                                                                                                                                                                                                    
        [                                                                                                                                                                                                           
            "source/**/*.cc",                                                                                                                                                                                       
            "source/**/*.h",                                                                                                                                                                                        
            "include/**/*.h",                                                                                                                                                                                       
        ],                                                                                                                                                                                                          
        exclude = ["source/exe/main.cc"],                                                                                                                                                                           
    ) + [                                                                                                                                                                                                           
        "source/common/version_generated.cc",                                                                                                                                                                       
    ],